### ISSUE TITLE
Useful information when .dill file is rejected

### ISSUE THREAD

**Issue by peter-ahe-google** [dart-lang/sdk#28692]
When I try to run this [dill file](https://github.com/dart-lang/sdk/files/761958/hello.zip) at a7f734325938ec3b985362cdd6f35135ac7eeef1, I get this error:

```
../../runtime/vm/object.cc: 15844: error: unreachable code
Dumping native stack trace for thread 307
Stack dump aborted because GetAndValidateIsolateStackBounds.
Abort trap: 6
```

The VM version is: `Dart VM version: 1.23.0-edge.d97472d231e4345d5207370a5fd1d62957c4da2a (Wed Feb  8 22:08:41 2017) on "macos_x64"`.

It would be nice to get a little more information.

I can see that the corresponding error is:

```
bool AbstractType::HasResolvedTypeClass() const {
  // AbstractType is an abstract class.
  UNREACHABLE();
  return false;
}
```

Unfortunately, the only way to debug this is using lldb, but that doesn't seem to work. lldb is so slow that it's virtually useless in the debug build, and just pretty slow in release build. I wonder if there's a build setting that needs tweaking?

**Comment by peter-ahe-google** [comment]
@mraleph

**Comment by peter-ahe-google** [comment]
I've found that I can use the option `--trace_type_finalization` to get some useful information.

**Comment by mraleph** [comment]
I filed a separate issue #28702 for LLDB + VM slowness caused by the profiler.

**Comment by peter-ahe-google** [comment]
Based on some info from @kmillikin, I've been pointed towards an issue with mixin applications and I'm now past this problem. And thanks to @mraleph, I can also debug.

I still think it would be great if we can get more information out of the type finalization, but I'm not blocked by this issue.

**Comment by jensjoha** [comment]
Was this "fixed" (i.e. is useful information now given) with @mraleph s commit?
